### PR TITLE
Clearer converter diagnostics

### PR DESCRIPTION
**Pull request by erikbaranowski**
#### PR Description

- Update errors to make it clearer they are converter errors and not configuration errors
- Tune some diagnostic messages and levels from various feedback
- Default promtail log level embedded in the static converter so unnecessary warnings don't show

#### PR Checklist

- [x] CHANGELOG.md updated
- [x] Documentation added
- [x] Tests updated
- [x] Config converters updated

### CODE REVIEW

File: converter/internal/common/http_client_config.go [hunk 1]
```diff
@@ -30,21 +30,10 @@ func ToHttpClientConfig(httpClientConfig *prom_config.HTTPClientConfig) *config.
...
-	}
-
-	if len(httpClientConfig.ProxyConnectHeader) > 0 {
-		diags.Add(diag.SeverityLevelError, "unsupported HTTP Client config proxy_connect_header was provided")
-	}
-
-	if httpClientConfig.TLSConfig.MaxVersion != 0 {
-		diags.Add(diag.SeverityLevelError, "unsupported HTTP Client config max_version was provided")
-	}
+	diags.AddAll(UnsupportedNotEquals(httpClientConfig.NoProxy, "", "HTTP Client no_proxy"))
```
Comment (mattdurham) [inline converter/internal/common/http_client_config.go:33]: nit: Should this be InvalidNotEquals since we changed the wording?

Comment (mattdurham) [reply converter/internal/common/http_client_config.go:33]: Hmmm, maybe not looking at it farther.

File: converter/internal/staticconvert/staticconvert.go [hunk 2]
```diff
@@ -129,6 +129,12 @@ func appendStaticPromtail(f *builder.File, staticConfig *config.Config) diag.Dia
 		promtailConfig.TargetConfig = logConfig.TargetConfig
 		promtailConfig.LimitsConfig = logConfig.LimitsConfig
 
+		// We are using the
+		err := promtailConfig.ServerConfig.Config.LogLevel.Set("info")
+		if err != nil {
+			panic("unable to default promtail log level from the static converter.")
```
Comment (mattdurham) [inline converter/internal/staticconvert/staticconvert.go:135]: verbiage seems off here, "unable to set default promtail..." instead?

**Review by mattdurham — APPROVED** [review]
Mostly verbiage but LGTM

File: converter/internal/staticconvert/staticconvert.go (hunk 2 above)
Comment (erikbaranowski) [reply converter/internal/staticconvert/staticconvert.go]: good call, updated

File: converter/internal/common/http_client_config.go (hunk 1 above)
Comment (erikbaranowski) [reply converter/internal/common/http_client_config.go:33]: I'm going to take a pass on these function names. They haven't kept up as the contents and location have evolved.

**Review by thampiotr — APPROVED** [review]

**Pull request merged**